### ISSUE TITLE
Prediction before fitting the data

### ISSUE THREAD

**Issue by sid548** [online-ml/river#363]
Hello,
I am having difficulty in understanding that why are we predicting first and then fit the data to model as opposed to batch learning where we will first fit the dataset and then it will be used for prediction.

**Comment by MaxHalford** [comment]
Hello! What's the question?

**Comment by sid548** [comment]
> Hello! What's the question?

from creme import compose
>>> from creme import linear_model
>>> from creme import metrics
>>> from creme import preprocessing

>>> model = compose.Pipeline(
...     preprocessing.StandardScaler(),
...     linear_model.LogisticRegression()
... )

>>> metric = metrics.Accuracy()

>>> for x, y in X_y:
...     y_pred = model.predict_one(x)      # make a prediction
...     metric = metric.update(y, y_pred)  # update the metric
...     model = model.fit_one(x, y)        # make the model learn

In the above code, model.predict_one() is used before model.fit_one(). But I guess it should be other way

**Comment by raphaelsty** [comment]
Hello, if I understood your issue correctly, you wish to have more details on the validation method adopted by Creme.

In a Batch learning environment: 

When you train a model on a set of training data, the model becomes a specialist on this data. You can't test your model on this data because it has learned it. You use what is called a validation set and/or test set to measure your model's ability to generalize to new data. In fact, what is interesting with the learning machine is to build a model that behaves correctly with non-labeled data. Usually, `cross-validation` is used to quantify the error of the model. 

In an online learning environment:

With Creme there is no need for a test or validation set. We measure the model's ability to generalize as we train. Before updating the model with a new observation, we quantify the model's capacity to generalize on this new observation. This process is called `progressive validation`.  The idea is simple, before updating your model, you check if your model is able to correctly predict the new observation.

**Comment by MaxHalford** [comment]
@sid548 I recommend that you go through the [From batch to online](https://creme-ml.github.io/examples/batch-to-online/) chapter in the user guide.

**Comment by sid548** [comment]
> Hello, if I understood your issue correctly, you wish to have more details on the validation method adopted by Creme.
> 
> In a batch learning environment:
> 
> When you train a model on a set of training data, the model becomes a specialist on this data. You can't test your model on this data because it has learned it. You use what is called a validation set and/or test set to measure your model's ability to generalize to new data. In fact, what is interesting with the machine learning is to build a model that behaves correctly with new data. Usually, `cross-validation` is used to quantify the error of the model.
> 
> In an online learning environment:
> 
> With Creme there is no need for a test or validation set. We measure the model's ability to generalize as we train. Before updating the model with a new observation, we quantify the model's capacity to generalize on this new observation. This process is called `progressive-validation`. The idea is simple, before updating your model, you check if your model is able to correctly predict the new coming observation.
> 
> Raphaël

Thanks a lot

**Comment by sid548** [comment]
> @sid548 I recommend that you go through the [From batch to online](https://creme-ml.github.io/examples/batch-to-online/) chapter in the user guide.
> 
> Regarding your comment:
> 
> > In the above code, model.predict_one() is used before model.fit_one(). But I guess it should be other way
> 
> Why do you assume this? I'm not sure you're correctly visualising what's happening.

Thanks a lot for sharing the article.

**Comment by julienneross** [comment]
Does any of you have or a link to any data prediction that is already program? I would love to study it. Thanks!
